File: 백준/Bronze/1110. 더하기 사이클/더하기 사이클.js
const input = require("fs").readFileSync("/dev/stdin").toString().trim();

function solution(N) {
  const original = parseInt(N);
  let current = original;
  let count = 0;

  while (true) {
    count += 1;
    const tens = Math.floor(current / 10);
    const units = current % 10;
    const sumDigits = tens + units;
    const newNumber = units * 10 + (sumDigits % 10);

    if (newNumber === original) {
      break;
    }
    current = newNumber;
  }
  return count;
}

console.log(solution(input));

File: 백준/Bronze/1110. 더하기 사이클/더하기 사이클.py
N = input()

def solution(N):
    original = int(N)
    current = original
    count = 0

    while True:
        count += 1
        tens = current // 10
        units = current % 10

        sum_digits = tens + units
        new_number = units * 10 + (sum_digits % 10)

        if new_number == original:
            break
        
        current = new_number
    return count

print(solution(N))